src/wp-admin/includes/class-wp-media-list-table.php

```diff
@@ -746,130 +746,104 @@ protected function get_default_primary_column_name() {
-						$delete_ays,
-						/* translators: %s: Attachment title. */
-						esc_attr( sprintf( __( 'Delete &#8220;%s&#8221; permanently' ), $att_title ) ),
-						__( 'Delete Permanently' )
-					);
-				}
-			}
+		$actions_raw['attach'] = array(
+			'markup'    => sprintf(
+				'<a href="#the-list" onclick="findPosts.open( \'media[]\', \'%s\' ); return false;" class="hide-if-no-js aria-button-if-js" aria-label="%s">%s</a>',
+				$post->ID,
+				/* translators: %s: Attachment title. */
+				esc_attr( sprintf( __( 'Attach &#8220;%s&#8221; to existing content' ), $att_title ) ),
+				__( 'Attach' )
+			),
+			'condition' => ( $this->detached && current_user_can( 'edit_post', $post->ID ) ),
+		);
 
-			if ( ! $this->is_trash ) {
-				if ( get_permalink( $post->ID ) ) {
-					$actions['view'] = sprintf(
-						'<a href="%s" aria-label="%s" rel="bookmark">%s</a>',
-						get_permalink( $post->ID ),
-						/* translators: %s: Attachment title. */
-						esc_attr( sprintf( __( 'View &#8220;%s&#8221;' ), $att_title ) ),
-						__( 'View' )
-					);
-				}
+		$actions_raw['copy'] = array(
+			'markup'    => sprintf(
+				'<span class="copy-to-clipboard-container"><button type="button" class="button-link copy-attachment-url media-library" data-clipboard-text="%s" aria-label="%s">%s</button><span class="success hidden" aria-hidden="true">%s</span></span>',
+				esc_url( wp_get_attachment_url( $post->ID ) ),
+				/* translators: %s: Attachment title. */
+				esc_attr( sprintf( __( 'Copy &#8220;%s&#8221; URL to clipboard' ), $att_title ) ),
+				__( 'Copy URL' ),
+				__( 'Copied!' )
+			),
+			'condition' => ( ! $this->is_trash && wp_get_attachment_url( $post->ID ) ),
+		);
 
-				$actions['copy'] = sprintf(
-					'<span class="copy-to-clipboard-container"><button type="button" class="button-link copy-attachment-url media-library" data-clipboard-text="%s" aria-label="%s">%s</button><span class="success hidden" aria-hidden="true">%s</span></span>',
-					esc_url( wp_get_attachment_url( $post->ID ) ),
-					/* translators: %s: Attachment title. */
-					esc_attr( sprintf( __( 'Copy &#8220;%s&#8221; URL to clipboard' ), $att_title ) ),
-					__( 'Copy URL' ),
-					__( 'Copied!' )
-				);
+		$actions_raw['download'] = array(
+			'markup'    => sprintf(
+				'<a href="%s" aria-label="%s" download>%s</a>',
+				esc_url( wp_get_attachment_url( $post->ID ) ),
+				/* translators: %s: Attachment title. */
+				esc_attr( sprintf( __( 'Download &#8220;%s&#8221;' ), $att_title ) ),
+				__( 'Download file' )
+			),
+			'condition' => ( wp_get_attachment_url( $post->ID ) ? true : false ),
+		);
 
-				$actions['download'] = sprintf(
-					'<a href="%s" aria-label="%s" download>%s</a>',
-					esc_url( wp_get_attachment_url( $post->ID ) ),
-					/* translators: %s: Attachment title. */
-					esc_attr( sprintf( __( 'Download &#8220;%s&#8221;' ), $att_title ) ),
-					__( 'Download file' )
-				);
+		foreach ( $actions_raw as $action_key_name => $action_link ) {
```

Comment:
I'd be a little concerned about performance impact of looping through the actions here.

Is it possible just to wrap `$actions['key'] = ...` in `if` statements containing the `condition`?

For example, rather than:
```php
$actions_raw['edit'] = array(
	'markup'    => sprintf(
		'<a href="%s" aria-label="%s">%s</a>',
		get_edit_post_link( $post->ID ),
		/* translators: %s: Attachment title. */
		esc_attr( sprintf( __( 'Edit &#8220;%s&#8221;' ), $att_title ) ),
		__( 'Edit' )
	),
	'condition' => ( ! $this->is_trash && current_user_can( 'edit_post', $post->ID ) ),
);
```
This:
```php
if ( ! $this->is_trash && current_user_can( 'edit_post', $post->ID ) ) {
	$actions['edit'] = sprintf(
		'<a href="%s" aria-label="%s">%s</a>',
		get_edit_post_link( $post->ID ),
		/* translators: %s: Attachment title. */
		esc_attr( sprintf( __( 'Edit &#8220;%s&#8221;' ), $att_title ) ),
		__( 'Edit' )
	);
}
```

Reply:
Thanks for your review, it makes sense to me. I have updated the PR.